src/src/ESP8266_xmodem_update.h

```diff
@@ -0,0 +1,113 @@
+#include <Arduino.h>
+#include "targets.h"
+#include "telemetry.h"
+#include "xmodem_server.h"
+
+extern bool LED;
+extern Telemetry telemetry;
+
+#if defined(Regulatory_Domain_AU_915) || defined(Regulatory_Domain_EU_868) || defined(Regulatory_Domain_FCC_915) || defined(Regulatory_Domain_AU_433) || defined(Regulatory_Domain_EU_433)
+#include "SX127xDriver.h"
+extern SX127xDriver Radio;
+#elif defined(Regulatory_Domain_ISM_2400)
+#include "SX1280Driver.h"
+extern SX1280Driver Radio;
+#else
+#error "Radio configuration is not valid!"
+#endif
+
+void array_to_string(byte array[], unsigned int len, char buffer[])
+{
+    for (unsigned int i = 0; i < len; i++)
+    {
+        byte nib1 = (array[i] >> 4) & 0x0f;
+        byte nib2 = (array[i] >> 0) & 0x0f;
+        buffer[i*2+0] = nib1  < 0xa ? '0' + nib1  : 'a' + nib1  - 0xa;
+        buffer[i*2+1] = nib2  < 0xa ? '0' + nib2  : 'a' + nib2  - 0xa;
+    }
+    buffer[len*2] = '\0';
+}
+
+void esp8266_xmodem_tx(struct xmodem_server *xdm, uint8_t byte, void *cb_data)
+{
+    (void)xdm;
+    Serial.write(byte);
+}
+
+void esp8266_xmodem_updater()
+{
+    Radio.End();
+
+    for (uint8_t i = 0; i < 5; i++)
+    {
+        Serial.println("ESP8266 XMODEM BOOTLOADER"); // tell script we got into bootloader
+    }
+
+    //uint32_t uploadSize = telemetry.BootloaderFilesize() ; // didn't seem to work, assign max size instead
```

Comment:
any idea why this doesn't work? Does it matter?

Reply:
I'm not sure, most the examples I found simply assign the max size then call updater.end(true) when the bytes have been sent